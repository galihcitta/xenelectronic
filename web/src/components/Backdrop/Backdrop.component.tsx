import React from 'react';

import { Overlay } from './Backdrop.styles';

export interface BackdropProps {
  show: boolean;
  handleClose: () => void;
}

export const Backdrop: React.FC<BackdropProps> = ({ show, handleClose }) => (
  <Overlay data-testid="Overlay" show={show} onClick={handleClose} />
);
