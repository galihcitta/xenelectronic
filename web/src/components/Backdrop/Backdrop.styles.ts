import styled from 'styled-components';

import { color } from '../../utils/styles';

export const Overlay = styled.div`
  display: ${(props: { show: boolean }) => (props.show ? 'block' : 'none')};
  width: 100%;
  height: 100%;
  position: fixed;
  z-index: 6;
  left: 0;
  top: 0;
  background: ${color.black};
  opacity: 0.65;
`;
