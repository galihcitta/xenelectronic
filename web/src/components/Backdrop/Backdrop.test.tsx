import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Backdrop, BackdropProps } from './Backdrop.component';

describe('<Backdrop />', () => {
  const mockHandleClose = jest.fn();

  const props: BackdropProps = {
    show: false,
    handleClose: mockHandleClose
  };

  const setup = (props: BackdropProps) => {
    const utils = render(<Backdrop {...props} />);
    return { ...utils };
  };

  it('renders correctly', () => {
    const { asFragment } = setup(props);
    expect(asFragment()).toMatchSnapshot();
  });

  it('contains display:none style rule in `Overlay` element', () => {
    const { getByTestId } = setup(props);
    expect(getByTestId('Overlay')).toHaveStyle('display: none');
  });

  describe('when `show` prop is true', () => {
    beforeEach(() => {
      props.show = true;
    });

    it('contains display:block style rule in `Overlay` element', () => {
      const { getByTestId } = setup(props);
      expect(getByTestId('Overlay')).toHaveStyle('display: block');
    });

    it('calls `handleClose` func', () => {
      const { getByTestId } = setup(props);
      fireEvent.click(getByTestId('Overlay'));
      expect(mockHandleClose).toBeCalledTimes(1);
    });
  });
});
