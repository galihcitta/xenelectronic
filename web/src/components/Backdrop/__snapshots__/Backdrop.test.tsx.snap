// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa dYtZKB"
    data-testid="Overlay"
  />
</DocumentFragment>
`;
